### PR TITLE
[SEDONA-395] Add RS_AsImage

### PR DESCRIPTION
**Pull request by iGN5117**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-395. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?
- Add RS_AsImage
- Add Jupyter wrapper for visualizing a df containing htmlString

## How was this patch tested?
- Added tests

## Did this PR include necessary documentation updates?

- Yes, I am adding a new API. I am using the [current SNAPSHOT version number](https://github.com/apache/sedona/blob/master/pom.xml#L29) in since `vX.Y.Z` format.

### CODE REVIEW

File: docs/api/sql/Raster-visualizer.md [hunk 1]
```diff
@@ -20,7 +22,31 @@ Output:
...
+Format: `RS_AsImage(raster: Raster, imageWidth: int = 200)`
+
+Spark SQL Example:
+
+```sql
+SELECT RS_AsImage(raster, 500) from rasters
+SELECT RS_AsImage(raster) from rasters
+```
+
+!!!Tip
```
Comment (jiayuasu) [inline docs/api/sql/Raster-visualizer.md:39]: This needs to be updated to SedonaUtils. Please give a short example about how to use SedonaUtils.display_image(df)

File: common/src/main/java/org/apache/sedona/common/raster/RasterOutputs.java [hunk 2]
```diff
@@ -204,4 +205,16 @@ private static String createPaddedMatrixStringFromInt(int[] values, int width) {
 
         return res.toString();
     }
+
+    public static String createHTMLString(GridCoverage2D raster, int imageWidth) throws IOException {
+        String rasterAsBase64 = asBase64(raster);
+        String imageString = String.format("data:image/png;base64,%s", rasterAsBase64);
+        String htmlString =  "<img src=\"" + imageString + "\" width=\"" + imageWidth + "\" />";//String.format("<img src=\\\"%s\\\" width=\\\"%d\\\" />", imageString, imageWidth);
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/raster/RasterOutputs.java:212]: Remove the comment please.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: docs/api/sql/Raster-visualizer.md [hunk 3]
```diff
@@ -20,7 +22,38 @@ Output:
...
+    RS_AsImage can be paired with SedonaUtils.display_image(df) wrapper inside a Jupyter notebook to directly print the raster as an image in the output, where the 'df' parameter is the dataframe containing the HTML data provided by RS_AsImage
+
+Example:
+```python
+df = sedona.read.format('binaryFile').load(DATA_DIR + 'raster.tiff').selectExpr(\"RS_FromGeoTiff(content) as raster\")
+htmlDF = df.selectExpr(\"RS_AsImage(raster, 500) as raster_image\")
+SedonaUtils.display_image(htmlDF)
+```
+
+![Output](../../image/DisplayImage.png)
```
Comment (jiayuasu) [inline docs/api/sql/Raster-visualizer.md:53]: This image is not committed in this PR. Please add it.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: docs/api/sql/Raster-visualizer.md (hunk 3 above)
Comment (iGN5117) [reply docs/api/sql/Raster-visualizer.md:53]: Added image. Sorry!

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**